pkg/hubble/parser/threefour/parser_test.go

```diff
@@ -946,7 +947,7 @@ func TestTraceNotifyLocalEndpoint(t *testing.T) {
 func TestDebugCapture(t *testing.T) {
 	f := &flowpb.Flow{}
 
-	parser, err := New(log, &testutils.NoopEndpointGetter, &testutils.NoopIdentityGetter, &testutils.NoopDNSGetter, &testutils.NoopIPGetter, &testutils.NoopServiceGetter)
+	parser, err := New(log, &testutils.NoopEndpointGetter, &testutils.NoopIdentityGetter, &testutils.NoopDNSGetter, &testutils.NoopIPGetter, &testutils.NoopServiceGetter, &testutils.NoopLinkGetter)
```

Comment:
nit:

Just a few lines below, we're obtaining the real `ifIndex` of the loopback device so we could test the old non-mockable getter. Now that we mock it, we could simplify it

The following could be replaced with `loIfIndex := 1`
https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/hubble/parser/threefour/parser_test.go#L956-L960

This if-statement could be removed:
https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/hubble/parser/threefour/parser_test.go#L991-L992

Reply:
:+1: nice cleanup, I've added a new patch on top to address this.